### PR TITLE
Disable further checking permissions if at least one is False

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
To avoid extra requests to the database, opa container and code execution on the server

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/iam/permissions.py [hunk 1]
```diff
@@ -2025,6 +2025,8 @@ def check_permission(self, request, view, obj) -> bool:
         for perm in permissions:
             result = perm.check_access()
             allow &= result.allow
+            if not allow:
+                break
```
Comment (SpecLad) [inline cvat/apps/iam/permissions.py:2029]: I don't think the `allow` variable is needed at all, given this change. You can just replace this with:

```python
if not result.allow:
    return False
```

and the last line with

```python
return True
```

Comment (bsekachev) [reply cvat/apps/iam/permissions.py:2029]: agree

Comment (SpecLad) [reply cvat/apps/iam/permissions.py:2029]: In fact... I don't think the `permissions` list is needed either. You can merge this loop with the previous loop, and just check each permission as it is created.

**Comment by nmanovic** [comment]
@bsekachev , let's add a line into changelog. The optimization looks good. Need double check. Do you have any numbers to estimate the enhancement?

**Review by SpecLad — APPROVED** [review]

**Pull request merged**